### PR TITLE
some locators updates and classes contract added

### PR DESCRIPTION
**Pull request by Amet015**

### CODE REVIEW

File: src/main/java/salesforce/ui/pages/AppPageFactory.java [hunk 1]
```diff
@@ -183,6 +201,7 @@ public static AbstractCampaignMembersPage getCampaignMembers() {
         return new CampaignMembersLightningPage();
     }
 
+
```
Comment (JuanCarlosMartinez4) [inline src/main/java/salesforce/ui/pages/AppPageFactory.java]: delete this line

**Review by JuanCarlosMartinez4 — APPROVED** [review]

File: src/main/java/salesforce/entities/Contract.java [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+ * ("Confidential Information"). You shall not disclose such Confidential
+ * Information and shall use it only in accordance with the terms of the
+ * license agreement you entered into with Jalasoft.
+ */
+
+package salesforce.entities;
+
+import salesforce.entities.constants.ContractConstant;
+import salesforce.utils.DateFormatter;
+
```
Comment (JuanCarlosMartinez4) [inline src/main/java/salesforce/entities/Contract.java]: delete this line please

File: src/main/java/salesforce/entities/constants/ContractConstant.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+package salesforce.entities.constants;
+
+/**
+ * Manages ContractConstant instance.
+ *
+ * @author Alan Escalera.
+ * @version 1.0 29 March 2020.
+ */
+public class ContractConstant {
```
Comment (JuanCarlosMartinez4) [inline src/main/java/salesforce/entities/constants/ContractConstant.java]: if this class is only to declare constant variable, it should be **public final class ...**

File: src/main/java/salesforce/entities/constants/ContractConstant.java [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    public static final String ACCOUNT = "Account";
+    public static final String STATUS = "Status";
+    public static final String START_DATE = "Start Date";
+    public static final String CONTRACT_TERM = "Contract Term";
+}
```
Comment (JuanCarlosMartinez4) [inline src/main/java/salesforce/entities/constants/ContractConstant.java:28]: add empty private constructor

File: src/main/java/salesforce/ui/pages/contract/ContractClassicPage.java [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+ */
+public class ContractClassicPage extends AbstractContractPage {
+    @Override
+    protected void waitUntilPageObjectIsLoaded() {
+
+    }
+
+    @Override
+    public AbstractNewOpportunityPage clickOnNewOpportunity() {
+        //To Do
```
Comment (JuanCarlosMartinez4) [inline src/main/java/salesforce/ui/pages/contract/ContractClassicPage.java:29]: delete this comment if the return is null

File: src/main/java/salesforce/ui/pages/newcontract/AbstractNewContractPage.java [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+ * license agreement you entered into with Jalasoft.
+ */
+
+package salesforce.ui.pages.newcontract;
+
+import salesforce.entities.Contract;
+import salesforce.entities.constants.ContractConstant;
+import salesforce.ui.pages.AbstractBasePage;
+import salesforce.ui.pages.contract.AbstractContractPage;
+
```
Comment (JuanCarlosMartinez4) [inline src/main/java/salesforce/ui/pages/newcontract/AbstractNewContractPage.java]: delete this line

File: src/main/java/salesforce/ui/pages/newcontract/NewContractClassicPage.java [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+    }
+
+    @Override
+    protected void setContractTerm(final String contractTerm) {
+        //To Do
+    }
+
+    @Override
+    public AbstractContractPage clickSaveContract() {
+        //To Do
```
Comment (JuanCarlosMartinez4) [inline src/main/java/salesforce/ui/pages/newcontract/NewContractClassicPage.java:49]: delete this comment if the return is  null

File: src/main/java/salesforce/ui/pages/newopportunity/AbstractNewOpportunityPage.java [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+ * license agreement you entered into with Jalasoft.
+ */
+
+package salesforce.ui.pages.newopportunity;
+
+import salesforce.entities.Opportunity;
+import salesforce.entities.constants.OpportunityConstant;
+import salesforce.ui.pages.AbstractBasePage;
+import salesforce.ui.pages.opportunity.AbstractOpportunityPage;
+
```
Comment (JuanCarlosMartinez4) [inline src/main/java/salesforce/ui/pages/newopportunity/AbstractNewOpportunityPage.java]: delete this line

File: src/main/java/salesforce/ui/pages/newopportunity/NewOpportunityClassicPage.java [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+    }
+
+    @Override
+    protected void setStage(final String stage) {
+        // To Do
+    }
+
+    @Override
+    public AbstractOpportunityPage clickSaveOpportunityButton() {
+        // To Do
```
Comment (JuanCarlosMartinez4) [inline src/main/java/salesforce/ui/pages/newopportunity/NewOpportunityClassicPage.java]: delete this comment if the return is null

File: src/main/java/salesforce/ui/pages/opportunity/AbstractOpportunityPage.java [hunk 10]
```diff
@@ -14,6 +14,7 @@
 import salesforce.ui.pages.AbstractBasePage;
 import salesforce.ui.pages.opportunity.opportunitycontactroles.AbstractContactRolesPage;
 import salesforce.ui.pages.opportunity.taskopportunity.AbstractTaskOpportunity;
+
```
Comment (JuanCarlosMartinez4) [inline src/main/java/salesforce/ui/pages/opportunity/AbstractOpportunityPage.java]: delete this line

File: src/main/java/salesforce/ui/pages/opportunity/AbstractOpportunityPage.java [hunk 11]
```diff
@@ -28,23 +29,42 @@
...
+     * Refresh the page.
+     */
+    public abstract void refreshPage();
+
     /**
      * Sets the form of new Campaign.
      *
      * @param opportunity entity.
-     * @param fields map.
+     * @param fields      map.
```
Comment (JuanCarlosMartinez4) [inline src/main/java/salesforce/ui/pages/opportunity/AbstractOpportunityPage.java]: delete unnecessary spaces

File: src/main/java/salesforce/ui/pages/opportunity/OpportunityClassicPage.java [hunk 12]
```diff
@@ -227,7 +248,7 @@ private String getContactNameText(final String contactName) {
      * Gets contact name.
      *
      * @param contactName value.
-     * @param rolName value.
+     * @param rolName     value.
```
Comment (JuanCarlosMartinez4) [inline src/main/java/salesforce/ui/pages/opportunity/OpportunityClassicPage.java]: delete unnecessary spaces

File: src/test/java/salesforcetest/steps/ContractSteps.java [hunk 13]
```diff
@@ -0,0 +1,139 @@
...
+import salesforce.entities.Contract;
+import salesforce.entities.Opportunity;
+import salesforce.ui.components.span.ToastUpdateObjectMessage;
+import salesforce.ui.pages.AppPageFactory;
+import salesforce.ui.pages.genericTabs.AbstractTabObjectsPage;
+import salesforce.ui.pages.newcontract.AbstractNewContractPage;
+import salesforce.ui.pages.newopportunity.AbstractNewOpportunityPage;
+import salesforce.ui.pages.oportunitieslist.AbstractOpportunityListPage;
+import salesforce.ui.pages.opportunity.AbstractOpportunityPage;
+
```
Comment (JuanCarlosMartinez4) [inline src/test/java/salesforcetest/steps/ContractSteps.java]: delete this line

File: src/test/java/salesforcetest/steps/OpportunityStep.java [hunk 14]
```diff
@@ -185,12 +185,12 @@ public void cratesNewCampaignWith(final Map<String, String> mapNewCampaign) {
...
     @When("I assign the Campaign to the opportunity")
     public void assignsTheCampaignToTheOpportunity(final Map<String, String> mapOpportunityEdit) {
-
         HashMap<String, String> mapOpportunity = new HashMap<>();
         mapOpportunity.putAll(mapOpportunityEdit);
-        opportunitiesPage = AppPageFactory.getTabObjectsPage();
-        opportunityPage = opportunitiesPage.selectObjectByName(context.getOpportunities().get(ARRAY_POSITION_FIRST)
+        abstractTabObjectsPage = AppPageFactory.getTabObjectsPage();
+        abstractTabObjectsPage.selectObjectByName(context.getOpportunities().get(ARRAY_POSITION_FIRST)
                 .getName());
```
Comment (JuanCarlosMartinez4) [inline src/test/java/salesforcetest/steps/OpportunityStep.java]: this line can be loacated on previous line

File: src/test/java/salesforcetest/steps/OpportunityStep.java [hunk 15]
```diff
@@ -256,9 +256,10 @@ public void selectOpportunity() {
     @And("I add new Task with")
     public void iAddNewTaskWith(final Map<String, String> mapTask) {
         mapNewTask = new HashMap<>(mapTask);
-        opportunitiesPage = AppPageFactory.getTabObjectsPage();
-        opportunityPage = opportunitiesPage.selectObjectByName(context.getOpportunities().get(ARRAY_POSITION_FIRST)
+        abstractTabObjectsPage = AppPageFactory.getTabObjectsPage();
+         abstractTabObjectsPage.selectObjectByName(context.getOpportunities().get(ARRAY_POSITION_FIRST)
                 .getName());
```
Comment (JuanCarlosMartinez4) [inline src/test/java/salesforcetest/steps/OpportunityStep.java]: if it possible put on previous line

File: src/main/java/salesforce/entities/Contract.java (hunk 2 above)
Comment (JuanCarlosMartinez4) [reply src/main/java/salesforce/entities/Contract.java]: x2

File: src/main/java/salesforce/entities/constants/ContractConstant.java (hunk 3 above)
Comment (JuanCarlosMartinez4) [reply src/main/java/salesforce/entities/constants/ContractConstant.java]: x2

File: src/main/java/salesforce/entities/constants/OpportunityConstant.java [hunk 16]
```diff
@@ -32,9 +32,9 @@
     public static final String MAIN_COMPETITOR = "MainCompetitor";
     public static final String DELIVERY_STATUS = "Delivery/InstallationStatus";
     public static final String DESCRIPTION = "Description";
-    public static final String CAMPAIGN_NAME = "Campaign Name";
     public static final String VALUE = "value";
 
+
```
Comment (JuanCarlosMartinez4) [inline src/main/java/salesforce/entities/constants/OpportunityConstant.java]: delete this line

File: src/main/java/salesforce/ui/pages/PageTransporter.java [hunk 17]
```diff
@@ -38,6 +38,8 @@ public PageTransporter() {
         urlLightningMap.put("Campaigns Page", PropertiesReader.getInstance().getSalesforceUrl()
                 .getCampaignsLightningUrl());
         urlLightningMap.put("Leads Page", PropertiesReader.getInstance().getSalesforceUrl().getLeadsLightningUrl());
+        urlLightningMap.put("Contracts Page", PropertiesReader.getInstance().getSalesforceUrl()
```
Comment (JuanCarlosMartinez4) [inline src/main/java/salesforce/ui/pages/PageTransporter.java:41]: You can use constants for this string **"Contracts Page"**

File: src/main/java/salesforce/ui/pages/campaignlist/CampaignListLightningPage.java [hunk 18]
```diff
@@ -51,9 +52,9 @@ public AbstractNewCampaignPage clickOnNewButton() {
 
     @Override
     public AbstractCampaignPage selectCampaignName(final String campaignName) {
-        String opportunityNameCss = String.format(NAME_CAMPAIGN, campaignName);
+        String opportunityNameXpath = String.format(NAME_CAMPAIGN, campaignName);
         nameCampaignSelected = WebDriverManager.getInstance().getWebDriver().findElement(By
-                .cssSelector(opportunityNameCss));
+                .xpath(opportunityNameXpath));
```
Comment (JuanCarlosMartinez4) [inline src/main/java/salesforce/ui/pages/campaignlist/CampaignListLightningPage.java:57]: if it is posible this line can be together with previous line

File: src/main/java/salesforce/ui/pages/contract/AbstractContractPage.java [hunk 19]
```diff
@@ -0,0 +1,30 @@
...
+ * ("Confidential Information"). You shall not disclose such Confidential
+ * Information and shall use it only in accordance with the terms of the
+ * license agreement you entered into with Jalasoft.
+ */
+
+package salesforce.ui.pages.contract;
+
+import salesforce.ui.pages.AbstractBasePage;
+import salesforce.ui.pages.opportunity.newopportunity.AbstractNewOpportunity;
+
```
Comment (JuanCarlosMartinez4) [inline src/main/java/salesforce/ui/pages/contract/AbstractContractPage.java:14]: delte this line

File: src/main/java/salesforce/ui/pages/contract/ContractLightningPage.java [hunk 20]
```diff
@@ -0,0 +1,43 @@
...
+ */
+
+package salesforce.ui.pages.contract;
+
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import salesforce.ui.pages.AppPageFactory;
+import salesforce.ui.pages.opportunity.newopportunity.AbstractNewOpportunity;
+
```
Comment (JuanCarlosMartinez4) [inline src/main/java/salesforce/ui/pages/contract/ContractLightningPage.java:17]: delete this line

File: src/main/java/salesforce/ui/pages/contractlist/ContractListLightningPage.java [hunk 21]
```diff
@@ -0,0 +1,75 @@
...
+
+    /**
+     * Clicks on moreActions button.
+     *
+     * @param contractNumber value.
+     */
+    private void clickOnMoreActions(final String contractNumber) {
+        webDriver.findElement(By.xpath(String.format(MORE_ACTIONS, contractNumber))).click();
+    }
+
```
Comment (JuanCarlosMartinez4) [inline src/main/java/salesforce/ui/pages/contractlist/ContractListLightningPage.java]: delete this line of codee

File: src/main/java/salesforce/ui/pages/newcontract/AbstractNewContractPage.java (hunk 6 above)
Comment (JuanCarlosMartinez4) [reply src/main/java/salesforce/ui/pages/newcontract/AbstractNewContractPage.java]: x2

File: src/main/java/salesforce/ui/pages/newcontract/NewContractLightningPopUp.java [hunk 22]
```diff
@@ -0,0 +1,119 @@
...
+    private static final String STATUS_TO_SELECT = "//div[@class='select-options']//a[@title='%s']";
+
+    private Actions actions = new Actions(webDriver);
+
+    @Override
+    protected void waitUntilPageObjectIsLoaded() {
+        webDriverWait.until(ExpectedConditions.visibilityOf(searchAccountsField));
+        webDriverWait.until(ExpectedConditions.elementToBeClickable(searchAccountsField));
+    }
+
```
Comment (JuanCarlosMartinez4) [inline src/main/java/salesforce/ui/pages/newcontract/NewContractLightningPopUp.java:57]: delete this line

File: src/main/java/salesforce/ui/pages/oportunitieslist/OpportunityListLightningPage.java [hunk 23]
```diff
@@ -25,19 +26,71 @@
...
+     */
+    private void clickOnMoreActions(final String name) {
+        webDriver.findElement(By.xpath(String.format(MORE_ACTIONS, name))).click();
+    }
+
     @Override
     protected void waitUntilPageObjectIsLoaded() {
         webDriverWait.until(ExpectedConditions.visibilityOfAllElements(opportunityTable));
     }
+
```
Comment (JuanCarlosMartinez4) [inline src/main/java/salesforce/ui/pages/oportunitieslist/OpportunityListLightningPage.java]: delete this line

File: src/main/java/salesforce/ui/pages/opportunity/AbstractOpportunityPage.java (hunk 10 above)
Comment (JuanCarlosMartinez4) [reply src/main/java/salesforce/ui/pages/opportunity/AbstractOpportunityPage.java]: x2

File: src/main/java/salesforce/ui/pages/opportunity/OpportunityClassicPage.java [hunk 24]
```diff
@@ -21,6 +21,7 @@
 import salesforce.ui.pages.opportunity.taskopportunity.AbstractTaskOpportunity;
 import salesforce.ui.pages.owner.OwnerEditClassicPage;
 import salesforce.utils.UtilSalesforce;
+
```
Comment (JuanCarlosMartinez4) [inline src/main/java/salesforce/ui/pages/opportunity/OpportunityClassicPage.java]: delete this line

File: src/main/java/salesforce/ui/pages/opportunity/OpportunityClassicPage.java (hunk 12 above)
Comment (JuanCarlosMartinez4) [reply src/main/java/salesforce/ui/pages/opportunity/OpportunityClassicPage.java]: x2

File: src/test/java/salesforcetest/steps/OpportunitySteps.java [hunk 25]
```diff
@@ -393,4 +398,50 @@ public void informationShouldBeDisplayed() {
                     actualOpportunityValues.get(key));
         }
     }
+
```
Comment (JuanCarlosMartinez4) [inline src/test/java/salesforcetest/steps/OpportunitySteps.java:399]: delete this line

File: src/test/java/salesforcetest/steps/OpportunitySteps.java [hunk 26]
```diff
@@ -393,4 +398,50 @@ public void informationShouldBeDisplayed() {
...
+     * the application should display a message.
+     *
+     * @param message value.
+     */
+    @Then("the application should display a message {string}")
+    public void theApplicationShouldDisplayAMessage(final String message) {
+        toastUpdateObjectMessage = new ToastUpdateObjectMessage();
+        String messageClosed = toastUpdateObjectMessage.getMessage();
+        org.junit.Assert.assertEquals(message, messageClosed);
+
```
Comment (JuanCarlosMartinez4) [inline src/test/java/salesforcetest/steps/OpportunitySteps.java]: delete this line

File: src/test/resources/features/Contract.feature [hunk 27]
```diff
@@ -0,0 +1,21 @@
...
+  @DeleteContacts @DeleteContract @DeletesAnOpportunity
+  Scenario: Close an opportunity created through a contract
+    Given  I create 1 "Basic" accounts
+    And I go to "Contracts Page"
+    And I create New Contract with
+      | Account       | [new Account] |
+      | Start Date    | TODAY         |
+      | Contract Term | 3             |
+    And I go to "Opportunities Page"
+#    And I create New Opportunity with
```
Comment (JuanCarlosMartinez4) [inline src/test/resources/features/Contract.feature]: delete this commented line

File: src/main/java/salesforce/ui/pages/contract/ContractClassicPage.java (hunk 5 above)
Comment (Amet015) [reply src/main/java/salesforce/ui/pages/contract/ContractClassicPage.java:28]: I can't because it will throw an error

Comment (JuanCarlosMartinez4) [reply src/main/java/salesforce/ui/pages/contract/ContractClassicPage.java:29]: read again the comment

File: src/main/java/salesforce/ui/pages/PageTransporter.java (hunk 17 above)
Comment (Amet015) [reply src/main/java/salesforce/ui/pages/PageTransporter.java:41]: All te class is this.

File: src/main/java/salesforce/ui/pages/campaignlist/CampaignListLightningPage.java (hunk 18 above)
Comment (Amet015) [reply src/main/java/salesforce/ui/pages/campaignlist/CampaignListLightningPage.java:57]: It is not possible

File: src/main/java/salesforce/ui/pages/contract/ContractClassicPage.java (hunk 5 above)
Comment (JuanCarlosMartinez4) [reply src/main/java/salesforce/ui/pages/contract/ContractClassicPage.java:29]: the observation was delete the comment

**Review by JuanCarlosMartinez4 — APPROVED** [review]

**Pull request merged**